documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -1013,3 +1027,228 @@ Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rul
+
+### <a name="R3019" />R3019 ArmResourcePropertiesBag
+**Output Message**: Top level property names should not be repeated inside the properties bag for ARM resource '{0}'. Properties [{1}] conflict with ARM top level properties. Please rename these.
+
+**Description**: Per [ARM guidelines](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md), top level properties should not be repeated inside th properties bag for ARM resources.
+
+**Why the rule is important**: [ARM guidelines](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md) enforce this rule. 
+
+**How to fix the violation**: Rename or remove conflicting property. Note that this may require a change on the service side and may cause a breaking change.
+
+**Examples**: 
+**Bad example**: "id" property is repeated in the model, as "Resource" already contains "id".
+```json
+"VersionedApplicationType": {
+  "description": "The versioned application type resource",
+  "properties": {
+    "properties": {
+      "id": {
+        "type": "string",
+        "description": "The name of the application type"
+      }
+    }
+  },
+  "allOf": [
+  {
+    "$ref": "#/definitions/Resource"
+  }
+  ]
+},
+"Resource":{  
+  "properties":{  
+    "id":{  
+      "readOnly":true,
+      "type":"string",
+      "description":"Fully qualified resource Id for the resource. Ex - /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{resourceProviderNamespace}/{resourceType}/{resourceName}"
+    },
+    "name":{  
+      "readOnly":true,
+      "type":"string",
+      "description":"The name of the resource"
+    },
+    "type":{  
+      "readOnly":true,
+      "type":"string",
+      "description":"The type of the resource. Ex- Microsoft.Compute/virtualMachines or Microsoft.Storage/storageAccounts."
+    }
+  },
+  "x-ms-azure-resource": true
+}
+```
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R3006" />R3006 BodyTopLevelProperties
+**Output Message**: Top level properties should be one of name, type, id, location, properties, tags, plan, sku, etag, managedBy, identity. Model definition '{0}' has extra properties ['{1}'].
+
+**Description**: Per [ARM guidelines](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md), top level properties of a resource should be only ones from the allowed set.
+
+**Why the rule is important**: [ARM guidelines](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md) enforce this rule. 
```

Comment:
I've kept only the URL per Deepak's suggestion. In terms of grammar, since guidelines is plural, I think it'd be "enforce".